core/templates/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -36,8 +36,80 @@ interface CkeditorCustomScope extends ng.IScope {
 
 angular.module('oppia').directive('ckEditor4Rte', [
   'CkEditorCopyContentService', 'ContextService', 'RteHelperService',
-  function(
-      CkEditorCopyContentService, ContextService, RteHelperService) {
+  function(CkEditorCopyContentService, ContextService, RteHelperService) {
+    /**
+     * Creates a CKEditor configuration.
+     * @param config CKEditor config to add to
+     * @param uiConfig Parameters to add to CKEditor config
+     * @param pluginNames Comma separated list of plugin names
+     * @param buttonNames Array of button names for RTE components
+     * @param extraAllowedContentRules additional allowed content rules for
+     * CKEDITOR.editor.filter
+     * @param sharedSpaces IDs of the page elements that will store the editor
+     * UI elements
+     */
+    const _createCKEditorConfig = function(
+        uiConfig: UiConfig,
+        pluginNames: string,
+        buttonNames: string[],
+        extraAllowedContentRules: string,
+        sharedSpaces: object
+    ): CKEDITOR.config {
+      // Language configs use default language when undefined.
+      let ckConfig: CKEDITOR.config = {
+        extraPlugins: 'pre,sharedspace,' + pluginNames,
+        startupFocus: true,
+        removePlugins: 'indentblock',
+        title: false,
+        floatSpaceDockedOffsetY: 15,
+        extraAllowedContent: extraAllowedContentRules,
+        sharedSpaces: sharedSpaces,
+        skin: (
+          'bootstrapck,' +
+          '/third_party/static/ckeditor-bootstrapck-1.0.0/skins/bootstrapck/'
+        ),
+        toolbar: [
+          {
+            name: 'basicstyles',
+            items: ['Bold', '-', 'Italic']
+          },
+          {
+            name: 'paragraph',
+            items: [
+              'NumberedList', '-',
+              'BulletedList', '-',
+              'Pre', '-',
+              'Blockquote', '-',
+              'Indent', '-',
+              'Outdent'
+            ]
+          },
+          {
+            name: 'rtecomponents',
+            items: buttonNames
+          },
+          {
+            name: 'document',
+            items: ['Source']
+          }
+        ]
+      };
+      if (uiConfig) {
```

Comment:
To reduce nesting, can you invert this into a guard clause? e.g.

```
if (!uiConfig) {
  return uiConfig;
}
...
```

Reply:
I changed to an inverted guard clause, thanks for the suggestion!